### PR TITLE
DBZ-7541 Add wait check to let database return online during SqlServerConnectorIT tests

### PR DESCRIPTION
**Pull request by mfvitale**
closes (hopefully :smile: ): https://issues.redhat.com/browse/DBZ-7541

This should avoid that tests will go long (4/5 hours) and then killed when sometime the database goes in a transition state.

### CODE REVIEW

**Review by Naros — COMMENTED** [review]
This seems reasonable to me @mfvitale, lets see how CI reacts.

**Comment by Naros** [comment]
Applied to main, thanks @mfvitale .

File: debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/SqlServerConnectorIT.java [hunk 1]
```diff
@@ -2967,10 +2971,29 @@ private void shouldStopRetriableRestartsAtConfiguredMaximum(SqlRunnable scenario
...
+                    .until(databaseIsOnline());
         }
     }
 
+    @NotNull
+    private Callable<Boolean> databaseIsOnline() {
+
+        return () -> connection.queryAndMap("select state from sys.databases where name = '" + TEST_DATABASE_2 + "'", rs -> {
+            rs.next();
+            System.out.println("DB status " + rs.getInt(1));
```
Comment (jpechane) [inline debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/SqlServerConnectorIT.java:2992]: @mfvitale Could you please replase `Syste.out.println` in this PRs with `Testing.Print`? Thanks

Comment (mfvitale) [reply debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/SqlServerConnectorIT.java:2992]: I'll address it with https://github.com/debezium/debezium/pull/5245

**Pull request merged**